### PR TITLE
Wrote the pipeline section

### PR DESCRIPTION
**Pull request by ClausWiingreen**

### CODE REVIEW

File: mainmatter/introduction/pipeline/index.tex [hunk 1]
```diff
@@ -0,0 +1,96 @@
+\section{The Graphics Pipeline} \label{sec:pipeline}
+\begin{sectionmeta}
+To understand the following sections it is recommended to gather an understanding of what the graphics pipeline is.
```
Comment (NotKyloRen) [inline mainmatter/introduction/pipeline/index.tex]: Recommend is odd, can that be changed please? (necessary)

File: mainmatter/introduction/pipeline/index.tex [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+The following section is based on a mix of the official documentation for the graphics pipelines for Direct3D and OpenGL \cite{khronos????pipeline, microsoft????pipeline}.
+\end{sectionmeta}
+
+The graphics pipeline can be compared to the \gls{CPU} pipeline in the way that it is a number of stages that each command goes through.
```
Comment (NotKyloRen) [inline mainmatter/introduction/pipeline/index.tex]: Too many "that"s, remove one and we're good to go

File: mainmatter/introduction/pipeline/index.tex [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+For graphics, that the command is instead an array of vertex data, such as position, texture coordinates, normals, and color, and unlike the \gls{CPU} pipeline, the relationship between the stages are not only one to one.
```
Comment (NotKyloRen) [inline mainmatter/introduction/pipeline/index.tex]: "That the command" is "the command" not enough?

File: mainmatter/introduction/pipeline/index.tex [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+    \path[arrow] (m.west|-t) -- (t);
+    \path[arrow] (m.west|-gs) -- (gs);
+    \path[arrow] (m.west|-ps) -- (ps);
+    \path[arrow, <->, shorten <=1pt] (m.west|-om) -- (om);
+  \end{tikzpicture}
+  \caption{A view of the graphics pipeline. The orange boxes are fixed functions, the blue boxes with rounded corners are shaders that the developer can replace with their own program, and the large green box is shared memory. Dashed shader stages are optional.}
+  \label{fig:pipeline01}
+\end{figure}
+
+The following paragraph goes through the different stages in the graphics pipeline, show in Figure \ref{fig:pipeline01}.
```
Comment (NotKyloRen) [inline mainmatter/introduction/pipeline/index.tex]: Shown* in Figure. (Show)

File: mainmatter/introduction/pipeline/index.tex [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+
+\paragraph{The Input Assembler} is the first stage. 
+It reads the user provided data and assembles the data into structures that it sends to the next stage in the pipeline.
+These structures include primitive types in the shader languages, such as floats, vectors, matrices, etc.
+
+\paragraph{The Vertex Shader} is the first interchangeable stage in the pipeline.
+Its main responsibility is to take a set of output from the Input Assembler and translate it into vertices in clip-space.
```
Comment (NotKyloRen) [inline mainmatter/introduction/pipeline/index.tex]: What's it secondary responsibility? If it doesn't have one, please use a different word.

File: mainmatter/introduction/pipeline/index.tex [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+Clip-space is a space where all verticies that should be displayed are within the x and y coordinates -1 to 1 and z coordinate 0 to 1 or -1 to 1 depending on the \gls{API}.
+It also calculate values that later stages in the pipeline uses.
```
Comment (NotKyloRen) [inline mainmatter/introduction/pipeline/index.tex]: Can the seconday responsibility be more explicitly explained? (what values does it calculate)

File: mainmatter/introduction/pipeline/index.tex [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+Its responsibility is to tessellate the primitive, which means dividing the primitive into multiple smaller primitives.
+In the figure it is simplified.
+The tessellation stage consists of three elements: two shader stages, and a fixed function between them.
+The first shader describes how much tessellation should be done, and how.
+The fixed function the generates the desired result and the last shader modifies the generated primitives.
+This modification could include moving vertices according to a mathematical formula, or a texture, to show details that was otherwise not present in the model.
+
+\paragraph{The Geometry Shader} takes in a single primitive and outputs any number of primitives.
+This is used to generate particles from points, or hair and grass on surfaces.
+It can also be used to mask some primitives if necessary.
```
Comment (NotKyloRen) [inline mainmatter/introduction/pipeline/index.tex:41]: Can an example be given of when you would want to mask primitives?

File: mainmatter/introduction/pipeline/index.tex [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+
+
+\paragraph{The Rasterizer Stage} takes the primitives and calculates which fragments on the screen it covers.
+Fragments in this context usually refers to pixels either on the screen or on a target render texture.
```
Comment (NotKyloRen) [inline mainmatter/introduction/pipeline/index.tex]: please get rid of "usually".

**Review by NotKyloRen — CHANGES_REQUESTED** [review]
Small changes need to be made before i'll approve.

File: mainmatter/introduction/pipeline/index.tex (hunk 2 above)
Comment (ClausWiingreen) [reply mainmatter/introduction/pipeline/index.tex]: <crime_rider> removes two </crime_rider>

File: mainmatter/introduction/pipeline/index.tex (hunk 7 above)
Comment (ClausWiingreen) [reply mainmatter/introduction/pipeline/index.tex]: No

Comment (ClausWiingreen) [reply mainmatter/introduction/pipeline/index.tex:41]: No

File: mainmatter/introduction/pipeline/index.tex (hunk 1 above)
Comment (ClausWiingreen) [reply mainmatter/introduction/pipeline/index.tex]: Ikke enig

**Review by NotKyloRen — DISMISSED** [review]
Changes done

**Review by NotKyloRen — APPROVED** [review]
Too much Danish flavour, but bearable.

**Pull request merged**